docs/abi.rst

```diff
@@ -0,0 +1,820 @@
+
+The second way to register a method is with the :any:`Router.method` decorator placed directly on a function. This way is equivalent to the first, but has some properties that make it more convenient for some scenarios. Below is an example equivalent to the prior one, but using the :code:`Router.method` syntax:
+
+.. code-block:: python
+
+    my_router = Router(
+        name="Calculator",
+        bare_calls=BareCallActions(
+            # allow this app to be created with a NoOp call
+            no_op=OnCompleteAction(action=Approve(), call_config=CallConfig.CREATE),
+            # allow standalone user opt in and close out
+            opt_in=OnCompleteAction(action=Approve(), call_config=CallConfig.CALL),
+            close_out=OnCompleteAction(action=Approve(), call_config=CallConfig.CALL),
+        ),
+    )
+
+    # NOTE: the first part of the decorator `@my_router.method` is the router variable's name
+    @my_router.method
+    def add(a: abi.Uint64, b: abi.Uint64, *, output: abi.Uint64) -> Expr:
+        """Adds the two arguments and returns the result.
+
+        If addition will overflow a uint64, this method will fail.
+        """
+        return output.set(a.get() + b.get())
+
+    @my_router.method(no_op=CallConfig.CALL, opt_in=CallConfig.CALL)
+    def addAndStore(a: abi.Uint64, b: abi.Uint64, *, output: abi.Uint64) -> Expr:
+        """Adds the two arguments, returns the result, and stores it in the sender's local state.
+
+        If addition will overflow a uint64, this method will fail.
+
+        The sender must be opted into the app. Opt-in can occur during this call.
+        """
+        return Seq(
+            output.set(a.get() + b.get()),
+            # store the result in the sender's local state too
+            App.localPut(Txn.sender(), Bytes("result", output.get())),
+        )
+
+Compiling a Router Program
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Now that we know how to add bare app call and method call handlers to a :code:`Router`, the next step is to compile the :code:`Router` into TEAL code.
+
+The :any:`Router.compile_program` method exists for this purpose. It combines all registered methods and bare app calls into two ASTs, one for the approval program and one for clear state program, then internally calls :any:`compileTeal` to compile these expressions and create TEAL code.
+
+In addition to receiving the approval and clear state programs, the :code:`Router.compile_program` method also returns a :code:`Contract` object `from the Python SDK <https://py-algorand-sdk.readthedocs.io/en/latest/algosdk/abi/contract.html#algosdk.abi.contract.Contract>`_. This object represents an `ARC-4 Contract Description <https://github.com/algorandfoundation/ARCs/blob/main/ARCs/arc-0004.md#contract-description>`_, which can be distributed to clients to enable them to call the methods on the contract.
```

Comment:
```suggestion
In addition to receiving the approval and clear state programs, the :code:`Router.compile_program` method also returns a :code:`Contract` object `from the Python SDK <https://py-algorand-sdk.readthedocs.io/en/latest/algosdk/abi/contract.html#algosdk.abi.contract.Contract>`_. This object represents an `ARC-4 Contract Description <https://arc.algorand.foundation/ARCs/arc-0004#contract-description>`_, which can be distributed to clients to enable them to call the methods on the contract.
```

suggest repointing to the foundation instead of github